File: euler3.rb
input=600851475143

factors ||= []
factor=2
remainder=input
while factor*factor < remainder
  #puts "Factor #factor"
  while remainder % factor == 0
    remainder = remainder/factor
    if not factors.last == factor
      factors << factor
    end
  end
  #puts "Remainder #remainder"
  #p factors
  factor += 1
end
if remainder != 1
  factors << remainder
end
p factors
p factors.last

File: golang/euler3.go
// The prime factors of 13195 are 5, 7, 13 and 29.
// What is the largest prime factor of the number 600851475143 ?

package main

import (
	"fmt"
	"os"
	"strconv"
)
import "./primefactorize"

func main() {
	input, err := strconv.Atoi(os.Args[1])
	if err != nil {
		fmt.Println(err)
	}
	factors := primefactorize.Primefactorize(input)
	max_factor := 0
	for k, _ := range factors {
		if k > max_factor {
			max_factor = k
		}
	}
	fmt.Println(factors)
	fmt.Println(max_factor)
}
